assets/src/blocks/Accordion/AccordionBlock.js

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable no-unused-vars */
+import { AccordionEditor } from './AccordionEditor';
+import { frontendRendered } from '../frontendRendered';
+import { CSS_VARIABLES_ATTRIBUTE } from '../CssVariablesAttribute';
+
+
+const BLOCK_NAME = 'planet4-blocks/accordion';
+
+export class AccordionBlock {
+  constructor() {
+    const { registerBlockType, unregisterBlockStyle, registerBlockStyle } = wp.blocks;
+    const { __ } = wp.i18n;
+
+    const attributes = {
+      accordion_title: {
+        type: 'string',
+        default: '',
+        selector: '.page-section-header'
+      },
+      accordion_description: {
+        type: 'string',
+        default: '',
+        selector: '.page-section-description'
+      },
+      accordion_rows: {
+        type: 'array',
+        default: [],
+        selector: '.accordion-content',
+      },
+      accordion_id: {
+        type: 'integer',
+        default: 1,
+      },
+      accordion_headline: {
+        type: 'string',
+        default: '',
+        selector: '.accordion-headline',
+      },
+      accordion_text: {
+        type: 'string',
+        default: '',
+        selector: '.accordion-text',
+      },
+      accordion_btn_text: {
+        type: 'string',
+        default: '',
+      },
+      accordion_btn_url: {
+        type: 'url',
+        default: '',
+      },
+      // accordion_icon: {
+      //   type: 'string',
+      //   default: '',
+      //   selector: '.accordion-icon'
+      // },
+      // accordion_style: { // Needed to convert existing blocks
+      //   type: 'string',
+      //   default: ''
+      // },
+
+      // },
+      // css_variables: CSS_VARIABLES_ATTRIBUTE,
+    };
+
+    registerBlockType( BLOCK_NAME, {
+      title: __( 'Accordion', 'planet4-blocks-backend' ),
+      icon: 'menu',
+      category: 'planet4-blocks-beta',
+      keywords: [
+        __(this.BLOCK_NAME),
+        __('faq')
+      ],
+      attributes,
+      deprecated: [
+        {
+          attributes,
+          save() {
+            return null;
+          },
+        }
+      ],
+      edit: ( { isSelected, attributes, setAttributes } ) => {
+
+        function addAccRow(){
```

Comment:
We could define this function (and also the following `removeAccRow`) in `AccordionEditor` file instead of here, so that we can write directly `edit: AccordionEditor` at line 83, which is much shorter 🙂

Reply:
I cleaned the code more and this could actually go in general. Currently as it works if block title & description are left empty they are hidden from the front end, so a person can add as much as wanted collapsible elements without the need of an array to hold them all. (and still looks on the frontend as if they are all one block)